Clarify link readonly and deletion policies (#8536)
geldata/gel: docs/reference/datamodel/links.rst

    The expression must be :ref:`Stable <ref_reference_volatility>`.

:eql:synopsis:`readonly := {true | false}`

>     If ``true``, the link is considered *read-only*.  Modifications of this link using ``update`` are prohibited once an object is created. All of the derived links **must** preserve the original *read-only* value. Changes to this link **can** occur if a link is deleted and the appropriate deletion policy allows it.

Reviewer: I'd hyperlink "derived" to an example/section in the docs where we show link inheritance / overload

>     If ``true``, the link is considered *read-only*.  Modifications of this link using ``update`` are prohibited once an object is created. All of the derived links **must** preserve the original *read-only* value. Changes to this link **can** occur if a link is deleted and the appropriate deletion policy allows it.

Reviewer 2: ```suggestion
    If ``true``, the link is considered *read-only*.  Modifications of this link using ``update`` are prohibited once an object is created. All of the derived links **must** preserve the original *read-only* value. Changes to this link **might** still occur if the target object is deleted and the deletion policy allows it.
```

Author: @aljazerzen I agree with everything except the change from "can" to "might still", since if you _have_ deleted a target object with the appropriate deletion policy, it _will_ update the link set. wdyt?

Author: I guess the language we're using here isn't completely clear: maybe "will" is better than "can" or "might still"?

Reviewer: Why do we have to use vague "can" / "might" instead of a concrete "will"?

Reviewer 2: I wanted to say that 'even on "required", it is not guaranteed that the value will not change'. So the value "might still change".

But "will" also conveys the same thing.
